build-support/pants_venv

```diff
@@ -11,26 +11,57 @@ REQUIREMENTS=(
   ${REPO_ROOT}/pants-plugins/3rdparty/python/requirements.txt
 )
 
+venv_dir_prefix="${REPO_ROOT}/build-support/pants_dev_deps"
+
+# N.B. this function would better belong in `virutalenv`, which focuses on discovering which Python
+# binary to use and thus is a natural fit for this function. However, this must be defined here
+# to avoid a circular dependency, as `pants_venv` calls `virtualenv` so cannot rely on a function
```

Comment:
This makes no sense. As you state, `pants_venv` call `virtualenv`; so `pants_venv` already depends on `virtualenv`. Having `pants_venv` depend on both executing `virtualenv` (its "main") as it does today and another function inside it seems completely sane.